tests/h/models/document/_document_test.py

```diff
@@ -411,46 +353,58 @@ def test_it_saves_all_the_document_metas(
-            mock_db_session,
-            annotation.target_uri,
-            document_meta_dicts,
-            [],
-            annotation.created,
-            annotation.updated,
+        self.assert_sub_items_stored(
+            Document,
+            caller,
+            field="document_meta_dicts",
+            data_items=document_meta_dicts,
+            storage_fn=create_or_update_document_meta,
         )
 
-        assert create_or_update_document_meta.call_count == 3
-        for document_meta_dict in document_meta_dicts:
-            create_or_update_document_meta.assert_any_call(
-                session=mock_db_session,
-                document=Document.find_or_create_by_uris.return_value.first.return_value,
-                created=annotation.created,
-                updated=annotation.updated,
-                **document_meta_dict,
-            )
-
-    def test_it_returns_a_document(
-        self, annotation, create_or_update_document_meta, Document, mock_db_session
-    ):
+    def assert_sub_items_stored(self, Document, caller, field, data_items, storage_fn):
         Document.find_or_create_by_uris.return_value.count.return_value = 1
 
-        result = update_document_metadata(
-            mock_db_session,
-            annotation.target_uri,
-            [],
-            [],
-            annotation.created,
-            annotation.updated,
+        caller(
+            session=sentinel.session,
+            created=sentinel.created,
+            updated=sentinel.updated,
+            **{field: data_items},
         )
 
-        assert result == Document.find_or_create_by_uris.return_value.first.return_value
+        assert storage_fn.call_count == len(data_items)
+
+        for item in data_items:
+            storage_fn.assert_any_call(
+                session=sentinel.session,
+                document=Document.find_or_create_by_uris.return_value.first.return_value,
+                created=sentinel.created,
+                updated=sentinel.updated,
+                **item,
+            )
 
     @pytest.fixture
-    def annotation(self):
-        # We can't use the factories here because our factories use the methods
-        # under test / being mocked here to create annotations
-        return mock.Mock(spec=models.Annotation())
+    def doc_uri_dicts(self):
+        return [
+            {
+                "uri": f"http://example.com/example_{i}",
+                "claimant": "http://example.com/claimant",
+                "type": "type",
+                "content_type": None,
+            }
+            for i in range(3)
+        ]
+
+    @pytest.fixture
+    def caller(self, db_session):
+        return functools.partial(
+            update_document_metadata,
+            session=sentinel.db_session,
+            target_uri=sentinel.target_uri,
+            created=sentinel.created,
+            updated=sentinel.updated,
+            document_meta_dicts=[],
+            document_uri_dicts=[],
+        )
```

Comment:
`functools.partial` is such a nice way of doing this. It prevents the usual pattern of defining your own function which duplicates all of the args.